src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -1,33 +1,86 @@
 package org.jenkinsci.plugins.ewm.steps;
 
 import com.google.inject.Inject;
+import hudson.FilePath;
 import hudson.model.Run;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
 import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.steps.model.ExternalWorkspace;
+import org.jenkinsci.plugins.ewm.strategies.DiskAllocationStrategy;
+import org.jenkinsci.plugins.ewm.strategies.MostUsableSpaceStrategy;
 import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
 import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
 
+import java.io.File;
 import java.util.List;
 
+import static java.lang.String.format;
+
 /**
- * TODO - To be added when I'll implement the step
+ * The execution of the {@link ExwsAllocateStep}.
  *
  * @author Alexandru Somai
  */
-public class ExwsAllocateExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+public class ExwsAllocateExecution extends AbstractSynchronousNonBlockingStepExecution<ExternalWorkspace> {
 
     @Inject(optional = true)
-    private transient ExwsAllocateStep exwsAllocateStep;
+    private transient ExwsAllocateStep step;
 
     @StepContextParameter
     private transient Run<?, ?> run;
+    @StepContextParameter
+    private transient TaskListener listener;
 
     @Override
-    protected Void run() throws Exception {
-        List<DiskPool> diskPools = exwsAllocateStep.getDescriptor().getDiskPools();
-        // TODO use globally defined disk pools
-        System.out.println(diskPools);
+    protected ExternalWorkspace run() throws Exception {
+        if (step.getUpstream() == null) {
```

Comment:
Maybe this condition os too strict. I can imagine a case when somebody starts an unrelated downstream job, which has its own workspace management logic